src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -628,6 +628,16 @@ internal int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancella
                         }
                     }
 
+                    string finalRefPeFilePath = Arguments.OutputRefFilePath;
+                    if (finalRefPeFilePath != null)
+                    {
+                        if (!CompileAndEmitRefAssembly(compilation, emitOptions.WithEmitMetadataOnly(true),
```

Comment:
>if (!CompileAndEmitRefAssembly(compilation, emitOptions.WithEmitMetadataOnly(true), [](start = 24, length = 83)

Perhaps we should assert that EmitOptions.EmitMetadataOnly is false if we get here?